File: FVResource.m
//
//  FVResource.m
//  ForkView
//
//  Created by Kevin Wojniak on 5/25/11.
//  Copyright 2011 Kevin Wojniak. All rights reserved.
//

#import "FVResource.h"
#import "FVResourcePriv.h"
#import "FVResourceFilePriv.h"

@implementation FVResource

@synthesize ident, name, dataSize, dataOffset, type, file;

- (void)setID:(uint16_t)anID
{
	ident = anID;
}

- (void)setName:(NSString *)aName
{
	if (name != aName) {
		[name release];
		name = [aName copy];
	}
}

- (void)setDataSize:(uint32_t)size offset:(uint32_t)offset
{
	dataSize = size;
	dataOffset = offset;
}

- (void)setType:(FVResourceType *)aType
{
	type = aType;
}

- (NSData *)data
{
	return [self.file dataForResource:self];
}

- (void)dealloc
{
	[name release];
	[super dealloc];
}

@end

File: FVResource.swift
//
//  FVResource.swift
//  ForkView
//
//  Created by Kevin Wojniak on 8/3/14.
//  Copyright (c) 2014 Kevin Wojniak. All rights reserved.
//

import Foundation

final public class FVResource: NSObject {
    @objc var ident: UInt16 = 0
    @objc var name: String = ""
    @objc var dataSize: UInt32 = 0
    @objc var dataOffset: UInt32 = 0
    @objc var type: FVResourceType? = nil
    @objc var file: FVResourceFile? = nil

    @objc var data: NSData? {
        return file?.dataForResource(resource: self)
    }
}
